kutt/kutt.go

```diff
@@ -0,0 +1,171 @@
+package kutt
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+)
+
+// URLDefinition represent URL definition from API
+type URLDefinition struct {
+	ID       string `json:"id"`
+	Target   string `json:"target"`
+	ShortURL string `json:"shortUrl"`
+}
+
+// API represent KuttIt API service
+type API struct {
+	httpClient   *http.Client
+	BaseURL      *url.URL
+	APIToken     string
+	CustomDomain string
+}
+
+// GetListURL will return list of URL from API
+func (api *API) GetListURL() ([]URLDefinition, error) {
+
+	rel := &url.URL{Path: "/api/url/geturls"}
+	u := api.BaseURL.ResolveReference(rel)
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("X-API-Key", api.APIToken)
+
+	api.httpClient = &http.Client{Timeout: 15 * time.Second}
```

Comment:
a 15 second timeout would be too much; this value would be good if this is configurable via ENV